pkg/cmd/server/api/types.go

```diff
@@ -14,12 +14,15 @@ const (
 )
 
 var (
-	KnownKubernetesAPILevels   = []string{"v1beta1", "v1beta2", "v1beta3", "v1"}
-	KnownOpenShiftAPILevels    = []string{"v1beta1", "v1beta3", "v1"}
-	DefaultKubernetesAPILevels = []string{"v1beta3", "v1"}
-	DefaultOpenShiftAPILevels  = []string{"v1beta3", "v1"}
-	DeadKubernetesAPILevels    = []string{"v1beta1", "v1beta2"}
-	DeadOpenShiftAPILevels     = []string{"v1beta1"}
+	KnownKubernetesAPILevels             = []string{"v1beta1", "v1beta2", "v1beta3", "v1"}
+	KnownOpenShiftAPILevels              = []string{"v1beta1", "v1beta3", "v1"}
+	KnownOpenShiftStorageVersionLevels   = []string{"v1beta1", "v1beta3", "v1"}
```

Comment:
Yes, that's fine.  Better to have at least a location where we describe
them.

On Wed, Oct 14, 2015 at 4:19 PM, Dan Mace <notifications@github.com> wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/4992#discussion_r42047054>:
>
> > @@ -14,12 +14,15 @@ const (
> >  )
> >
> >  var (
> > -	KnownKubernetesAPILevels   = []string{"v1beta1", "v1beta2", "v1beta3", "v1"}
> > -	KnownOpenShiftAPILevels    = []string{"v1beta1", "v1beta3", "v1"}
> > -	DefaultKubernetesAPILevels = []string{"v1beta3", "v1"}
> > -	DefaultOpenShiftAPILevels  = []string{"v1beta3", "v1"}
> > -	DeadKubernetesAPILevels    = []string{"v1beta1", "v1beta2"}
> > -	DeadOpenShiftAPILevels     = []string{"v1beta1"}
> > +	KnownKubernetesAPILevels             = []string{"v1beta1", "v1beta2", "v1beta3", "v1"}
> > +	KnownOpenShiftAPILevels              = []string{"v1beta1", "v1beta3", "v1"}
> > +	KnownOpenShiftStorageVersionLevels   = []string{"v1beta1", "v1beta3", "v1"}
>
> I can do this but I'm not sure what the upstream equiv of our version
> constants are. Should I just hard-code them like the others for now?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4992/files#r42047054>.
>